Library/Homebrew/token_auditor.rb

```diff
@@ -0,0 +1,40 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  class TokenAuditor
+    sig { returns(String) }
+    attr_reader :token
+
+    sig { params(token: String).void }
+    def initialize(token)
+      @token = token
+    end
+
+    sig { returns(T::Array[String]) }
+    def errors
+      errors = []
+
+      errors << "uppercase letters" if token.match?(/[A-Z]/)
+      errors << "whitespace" if token.match?(/\s/)
+      errors << "non-ASCII characters" unless token.ascii_only?
+      errors << "double hyphens" if token.include?("--")
+
+      # A bunch of formulae contain these:
+      # errors << "underscores" if token.include?("_")
+      # errors << "plus symbols" if token.include?("+")
```

Comment:
Maybe accept the formula or cask itself in the initializer and then check these only if it's a cask?

Reply:
Given we allow `_` and `+` for formula names, I don't really see a good reason to disallow it for casks.

Reply:
I don't feel strongly at the moment, but the comments should be removed if this is the plan

Reply:
Yes, I'll remove the comments before merging. We should check if casks should be renamed (`plus` -> `+`).